### ISSUE TITLE
Wrong field focus when typing

### ISSUE THREAD

**Issue by bdillahu** [duplicati/duplicati#2974]
- [X] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.16_canary_2018-01-18
- **Operating system**: MacOS High Sierra 10.13.2
- **Backend**: Google Drive

## Description

When entering a new configuration, I added an advanced option for "Retention Policy" - when typing in the field after each character the focus jumps out of the field. To type more you have to re-click in the field.

## Steps to reproduce

1. Add advanced option (retention policy)
2. Type multiple characters in field
3. ...

- **Actual result**:
After each character focus jumps out of field
- **Expected result**:
Ability to type multiple characters

**Comment by Pectojin** [comment]
Hey @kenkendk I think I found that thing that you fixed with the `$index` tracking, which you couldn't remember why you added ;) (#2946).

The issue happens consistently on v2.0.2.16_canary (where that PR was merged into) but not on v2.0.2.15_canary, and that PR was the only change to the [advancedoptionseditor.html](https://github.com/duplicati/duplicati/blame/c05ad799686c75f2c4eabd246f5558d09906c86e/Duplicati/Server/webroot/ngax/templates/advancedoptionseditor.html) file for a full year.

Without the `$index` tracking Angular seems to be replacing the entire `item` HTML element and reinserting it, on every character input, causing the loss of focus. With the `$index` tracking the element isn't replaced.

I think the reason is that when tracking without the index, it'll resort them every time the object is changed (even though index isn't changed). But the index tracking was also causing problems.

I'm looking into some options.
